src/librustc/metadata/creader.rs

```diff
@@ -350,12 +350,19 @@ impl<'a> CrateReader<'a> {
     fn is_staged_api(&self, data: &[u8]) -> bool {
         let attrs = decoder::get_crate_attributes(data);
         for attr in &attrs {
-            if &attr.name()[..] == "staged_api" {
-                match attr.node.value.node { ast::MetaWord(_) => return true, _ => (/*pass*/) }
+            if attr.name() == "feature" {
+                if let Some(metas) = attr.meta_item_list() {
+                    for meta in metas {
+                        if let ast::MetaWord(ref name) = meta.node {
+                            if &name[..] == "staged_api" {
+                                return true
```

Comment:
This overloads the meaning of `[feature(staged_api)]` in a way which is really confusing; feature attributes aren't supposed to have any semantic meaning.  On master, `[feature(staged_api)]` means "allow the use of unstable attributes related to staged APIs".  `[staged_api]` means "apply stability rules to users of this crate".  

Granted, the distinction is sort of useless at the moment, but we could, for example, decide that we shouldn't serialize feature attributes in the crate metadata because they don't affect users of a crate, or add `[feature(all)]` to turn off feature gate checking for a crate.